Compile error: fatal error: nccl.h: No such file or directory

Error:

```
In file included from /home/zhaochengduo/program/temp_master_paddle/Paddle/paddle/platform/enforce.h:39:0,
                 from /home/zhaochengduo/program/temp_master_paddle/Paddle/paddle/platform/gpu_info.cc:19:
/home/zhaochengduo/program/temp_master_paddle/Paddle/paddle/platform/dynload/nccl.h:18:18: fatal error: nccl.h: No such file or directory
 #include <nccl.h>
                  ^
compilation terminated.
make64[2]: *** [paddle/platform/CMakeFiles/gpu_info.dir/gpu_info.cc.o] Error 1
make64[1]: *** [paddle/platform/CMakeFiles/gpu_info.dir/all] Error 2
make64[1]: *** Waiting for unfinished jobs....
```
Environmental：

    host system: centos, version: 6.3
    CUDA 7.5, libcudnn.so.5.1.3  or CUDA 8.0, libcudnn.so.6.0.21

I change 
```
-#include <nccl.h>
 #include <mutex>
+#include "nccl.h"
```
but there are another mistake:
```
In file included from /home/luotao02/Paddle/paddle/platform/gpu_info.cc:19:0:
/home/luotao02/Paddle/paddle/platform/enforce.h:180:5: error: ‘paddle::platform::throw_on_error’ declared as an ‘inline’ variable
     ncclResult_t stat, const Args&... args) {
     ^
/home/luotao02/Paddle/paddle/platform/enforce.h:180:5: error: template declaration of ‘typename std::enable_if<(sizeof (Args ...) != 0), void>::type paddle::platform::throw_on_error’
/home/luotao02/Paddle/paddle/platform/enforce.h:180:5: error: ‘ncclResult_t’ was not declared in this scope
/home/luotao02/Paddle/paddle/platform/enforce.h:180:24: error: expected primary-expression before ‘const’
     ncclResult_t stat, const Args&... args) {
                        ^
make64[2]: *** [paddle/platform/CMakeFiles/gpu_info.dir/gpu_info.cc.o] Error 1
make64[1]: *** [paddle/platform/CMakeFiles/gpu_info.dir/all] Error 2
make64[1]: *** Waiting for unfinished jobs....
```

--- a/cmake/external/nccl.cmake
+++ b/cmake/external/nccl.cmake
@@ -2,7 +2,7 @@ INCLUDE(ExternalProject)
 
 SET(NCCL_SOURCE_DIR ${THIRD_PARTY_PATH}/nccl)
 
-INCLUDE_DIRECTORIES(${NCCL_SOURCE_DIR}/src/extern_nccl)
+INCLUDE_DIRECTORIES(${NCCL_SOURCE_DIR}/src/extern_nccl/src)
 
 
 if(WITH_DSO)

--- a/paddle/operators/math/im2col.h
+++ b/paddle/operators/math/im2col.h
@@ -75,7 +75,8 @@ class Im2ColFunctor {
   void operator()(const platform::DeviceContext& context,
                   const framework::Tensor& im, framework::Tensor& col,
                   int stride_height, int stride_width, int padding_up,
-                  int padding_down, int padding_left, int padding_right);
+                  int padding_down, int padding_left = 0,
+                  int padding_right = 0);
 };
 
 template <ColFormat Format, typename Place, typename T>
@@ -84,7 +85,7 @@ class Col2ImFunctor {
   void operator()(const platform::DeviceContext& context, framework::Tensor& im,
                   const framework::Tensor& col, int stride_height,
                   int stride_width, int padding_up, int padding_down,
-                  int padding_left, int padding_right);
+                  int padding_left = 0, int padding_right = 0);
 };
 
 }  // namespace math

